app/views/mw_interactives/_show.html.haml

```diff
@@ -35,11 +35,27 @@
 
 :javascript
   $(document).ready(function() {
-    $('##{dom_id_for(interactive, :click_to_play)}').click(function() {
-      $(this).css('display','none');
-      $(this).removeClass('shown')
+    var loadIframe = function(component) {
+      $(component).css('display','none');
+      $(component).removeClass('shown')
       $('##{dom_id_for(interactive, :interactive_image)}').css('display','none');
       $('#interactive_#{interactive.id}').attr('src','#{interactive.url}');
-      InteractiveManager.register($(this).parent('.interactive-container'))
+      InteractiveManager.register($(component).parent('.interactive-container'));
+    };
+    $('##{dom_id_for(interactive, :click_to_play)}').click(function() {
+      if ('#{interactive.preflight_url}' !== '') {
+        // pop up a pre-flight window. When it closes itself, load the iframe.
+        var component = this,
+            win = window.open('#{interactive.preflight_url}', 'preflight_window', 'height=120,width=120,location=0,menubar=0,resizable=0,scrollbars=0,status=0,titlebar=0,toolbar=0');
+        win.blur(); // pop under the main window
+        var poller = setInterval(function() {
+          if (win.closed !== false) {
```

Comment:
Yeah, the page I've set up for testing in the Document Server just uses javascript to close the window immediately:
```
<script>
  window.close();
</script>
```